File: frontend/src/config/platformBrand.ts
export const platformBrand = {
  chineseName: "全球胡杨林生态系统保护数据共享平台",
  englishName:
    "Global Populus euphratica Forest Ecosystem Conservation Data Sharing Platform",
  shortName: "GPEDSP",
  edition: "GPEDSP · WebGIS Research Edition",
} as const;

const legacyPlatformNames = new Set([
  "中亚胡杨林生态系统保护数据共享平台",
  "中亚胡杨林生态保护数据共享平台",
  "中亚胡杨林生态数据共享平台",
  "中亚胡杨生态系统保护数据共享平台",
  "中亚胡杨生态数据门户",
]);

export function resolvePlatformName(name?: string | null) {
  const normalized = name?.trim() ?? "";
  if (!normalized || legacyPlatformNames.has(normalized)) {
    return platformBrand.chineseName;
  }
  return normalized;
}

export function applyPlatformDocumentTitle(name?: string | null) {
  const title = resolvePlatformName(name);
  document.title = title;
  return title;
}

File: backend/apps/core/platform_brand.py
from __future__ import annotations

PLATFORM_CHINESE_NAME = "全球胡杨林生态系统保护数据共享平台"
PLATFORM_ENGLISH_NAME = (
    "Global Populus euphratica Forest Ecosystem Conservation Data Sharing Platform"
)
PLATFORM_ABBREVIATION = "GPEDSP"
PLATFORM_EDITION = "GPEDSP · WebGIS Research Edition"

LEGACY_PLATFORM_NAMES = frozenset(
    {
        "中亚胡杨林生态系统保护数据共享平台",
        "中亚胡杨林生态保护数据共享平台",
        "中亚胡杨林生态数据共享平台",
        "中亚胡杨生态系统保护数据共享平台",
        "中亚胡杨生态数据门户",
    }
)


def canonicalize_platform_name(value: object) -> str:
    """Upgrade known legacy brand names while preserving custom deployment names."""
    name = str(value or "").strip()
    return PLATFORM_CHINESE_NAME if name in LEGACY_PLATFORM_NAMES else name
